### ISSUE TITLE
Cache Duration in Cake 2.x

### ISSUE THREAD

**Issue by garethellis36** [beporter/CakePHP-EnvAwareness#7]
Hi Brian,

I just noticed a bug in your 2.x core.php. It jumped out at me because I had the same bug in my production app until a few months ago!

Line 25:
`Configure::write("debug", 0);`

Line 334:

```
// In development mode, caches should expire quickly.
$duration = '+999 days';
if (Configure::read('debug') > 0) {
    $duration = '+10 seconds';
}
```

Environment overrides are not called until later in the file. Therefore, when the value of debug is checked, it is always zero. As a result, dev environments will have long caches.

I am working on implementing your system into our app at the moment and my solution to this is simply to define Cache.Duration as a configuration key in core-{$env} and move the app override code further up core.php.

**Comment by cwbit** [comment]
Using a configuration key is actually more in line with the newer Cake 3 strategy of `app-{$env}`, too - so, good call!

**Comment by beporter** [comment]
Yep, Chris beat me to it. I'm working on a commit that demonstrates the modifications to make for Cake 2 to handle this. I'll post when I've pushed it.

**Comment by beporter** [comment]
Okay, so this is indeed another excellent point, and is indeed another example where we need to make Cake 2 more like Cake 3 by moving configs into `Configure`.

The basic idea is instead of calling `Cache::config()` in the middle of your core.php file directly, you'll set Configure values first, then _after_ loading env-specific configs, you'll configure the Caches.

I just pushed a commit for the sample Cake 2 app that demonstrates one possible solution: 8367acd

The core of that change is [right here](https://github.com/beporter/CakePHP-EnvAwareness/commit/8367acde560dde2bb322a713f36e3699f0510e43#diff-48ebaa5eb611b8d81090e9fc479aefb2R481). This is a little clunkier than how I handle the Database and Email configs (which have their own classes in which to do the necessary dirty work) but it gets the job done with minimal duplication.
